### DISCUSSION TITLE
+++ HELP NEEDED +++ Any code snippet to let elementor vertical unfold menu behave like an accordion?

### DISCUSSION THREAD

**Discussion by formgarten** [elementor/elementor#21439] [Developers Community]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

I'm working on a customer project and I don't like the vertical elementor menu. It's possible to open all navigation items of the second depth at the mopment. The navigation menu is very long in this way. 

### Describe the solution you'd like

My customer and I would like to have a possibility of an accordion, where only the menu items of the second depth are shown if you have selected the associated navigation point of the first depth while closing all others. 

### Describe alternatives you've considered

using another plugin called unlimited elements

### Additional context

The official elementor support again can't help me out with a code snippet as they're not allowed to. There is an 3rd party plugin for that (unlimited elements), but I don't want to install another plugin just because of this "little" issue. Any ideas?

**Comment by m1ga** [comment]
wouldn't it be the easiest way to use the plugin if it has the features you want? Overwriting the normal behavior sounds more complicated.
Otherwise you could create a [own widget](https://developers.elementor.com/docs/widgets/simple-example/) and use `wp_nav_menu` to output the menu. Then you can add your own CSS/JS to get the behavior you want to have.

**Reply by formgarten** [reply]
I think that would be a very useful addition to the menu widget and creating an widget I can't do that -> graphicdesigner. I really hope there's an option with with some custom code. 

But regarding elementor, I don't have much hope that this option will be added. I remember the accordion widget itself, where there were requests to show the accordion completely closed when the page was called up. The request has often been around for many years and nothing has happened to date.

**Reply by m1ga** [reply]
The accordion part is a different topic. You can either add a first blank item and hide that with CSS or use a simple JS like
```
document.querySelectorAll(".elementor-accordion .elementor-active").forEach(function(item) {
	item.classList.remove("elementor-active");
	if (item.classList.contains("elementor-tab-content")) item.style.display = "none";
})
```
to close all open items (without adding a blank first one). Accordion extends tabs and as you can see here:
https://github.com/elementor/elementor/blob/2cd3d50d8d87edde9cd3c820f5a0f8b1f2c4f24f/assets/dev/js/frontend/handlers/base-tabs.js#L42
it will default to "1" if there is no `activeItemIndex` (from the tabgroup). 

>  I really hope there's an option with with some custom code.

There is, but you have to code it or pay someone to do it :) Otherwise: just use `unlimited elements` if that menu is exactly what you want. External plugins are there for a reason so not everything has to be in Elementor itself.

**Reply by formgarten** [reply]
I know that workaround of the accordion. Thanks

**Reply by m1ga** [reply]
added a PR to have it as a switch in Elementor https://github.com/elementor/elementor/pull/21445

**Comment by formgarten** [comment]
Just in order to show it. In unlimited elements the option of the menu just looks like this. It's possible to decide wether to let all sub menu items be shown or not. Unfortunately this widget has some other bugs why I don't want to use it.

<img width="373" alt="Bildschirm­foto 2023-02-23 um 12 59 38" src="https://user-images.githubusercontent.com/55788402/220899765-c8a22f84-32fe-4b1c-a96e-38cdd041bbcc.png">

**Reply by m1ga** [reply]
https://user-images.githubusercontent.com/4334997/220907615-2f4bd4fc-70e4-454f-a748-8aec7f48f18b.mp4

but that would be a plugin and you said you don't want to install one.

But as you see: the functionality is really simple (`wp_nav_menu()` with some JS) but it doesn't look good. So that would take some hours to have some styling and options you can chose in Elementor. And that would be more than a Lifetime starter license of UnlimitedElements. You could try fiverr and get someone to create that widget for 50$ (or less) or even ChatGPT :smile: 


![Screenshot_20230223_134629](https://user-images.githubusercontent.com/4334997/220910332-f865dc36-5895-4c07-a2bc-de5ede536214.png)
not tried it but it combination with the complete "simple example" I've linked in the first reply this would work. 


another option: look at other "normal" wordpress vertical menus. There are plenty out there.

**Reply by formgarten** [reply]
Maybe I should talk to a developer friend of me :)) Your example really seems not to be that difficult. Well maybe we'll see that nonetheless inside the elementor menu widget.

**Reply by formgarten** [reply]
I'd like to finish: of course you can't fit every single wish into a plugin, but I think the navigation menu widget is pretty much one of the most important widgets in the whole elementor portfolio. And in my opinion, there has been a lot of room for improvement here for years, not so much the big, super special things, but rather the small details that have always been missing. And I find that my desire belongs there.

Thanks for your support 👏

**Reply by m1ga** [reply]
totally agree! Especially the vertical menu is not that useful :smile: There will be the new "mega menu" in the next release so that might already be a solution for all those issues as you can build the content more flexible

**Comment by formgarten** [comment]
By the way (yes, another topic): because of the first accordion item above I used the css workaround. Placing it code snippet plugin your js snippet does nothing for me. Where did you place it?

**Reply by m1ga** [reply]
Not sure if I understand the question. If you use the CSS fix you don't need the JS version. Not using any of them, only tested them for this thread :smile: I use [my plugin](https://wordpress.org/plugins/accordion-title-for-elementor/) or internal plugins.

But for the test I've used it this way:
* CSS: either in the custom CSS section (Pro) or just the default "additional CSS" section of wordpress or your theme (I mostly use the free version). Pro: works right away, even in the editor, Con: additional item.
* JS: I've used `WPCode – Insert Headers and Footers` plugin to add custom code (again: free Elementor :smile: ). There I've put it in the footer and use a simple timeout around it to make sure the accordion is initialized already. Think it was 250ms which was enough. I'm sure there is a better way to wait for the accordion JS to be ready but this worked. Pro: just the content you want, Con: timeout stuff, might flash a bit when its above the fold

**Reply by formgarten** [reply]
Ah, okay, I just copied it into WP Code (using it, too) without a timeout around it. That tells it all
